sound/soc/sof/topology.c

```diff
@@ -3046,8 +3049,28 @@ static void sof_complete(struct snd_soc_component *scomp)
 static int sof_manifest(struct snd_soc_component *scomp, int index,
 			struct snd_soc_tplg_manifest *man)
 {
-	/* not currently parsed */
-	return 0;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+
+	/* backward compatible with tplg without ABI info */
+	if (!man->priv.size) {
+		dev_dbg(sdev->dev, "No topology ABI info\n");
+		return 0;
+	}
+
+	if (man->priv.size == SOF_TPLG_ABI_SIZE) {
+		dev_info(sdev->dev,
+			 "Topology: ABI %d:%d:%d Kernel ABI %d:%d:%d\n",
+			 man->priv.data[0], man->priv.data[1],
+			 man->priv.data[2], SOF_ABI_MAJOR, SOF_ABI_MINOR,
+			 SOF_ABI_PATCH);
+		if (man->priv.data[0] == SOF_ABI_MAJOR &&
```

Comment:
@bardliao I am having second thoughts here. If we update the kernel to a new version, this would prevent an older topology file from being used? That's not good. We would want to allow this file to be used and it'd be up to the kernel to deal with changes in backwards compatible ways. Should it be <= SOF_ABI_MAJOR?

Reply:
@plbossart good point but in that then we cannot check the major and minor version individually.

Reply:
yes, we should use 
````c
if (SOF_ABI_VER(man->priv.data[0], man->priv.data[1], man->priv.data[2]) <= SOF_ABI_VERSION) 
````
right?

Reply:
@plbossart yes, thats right. Do you want me to amend Bard's PR and submit the change?